### PR TITLE
API module splitting

### PR DESCRIPTION
**Pull request by dcoutts**
This is a follow-up to #2040 that:
* actually uses the modules that were split out
* changes the exact details of what was already split out
* splits out more things

### CODE REVIEW

File: cardano-api/src/Cardano/Api/SerialiseCBOR.hs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE DefaultSignatures #-}
+
+-- | CBOR serialisation
+--
+module Cardano.Api.SerialiseCBOR
```
Comment (intricate) [inline cardano-api/src/Cardano/Api/SerialiseCBOR.hs:5]: Ah, I might prefer `SerialiseCbor` or `Serialise.Cbor`, but we've already named the type `SerialiseAsCBOR` so this is fine.

**Review by intricate — APPROVED** [review]
Looks good! 

Really nice series of `git` commits, btw.

File: cardano-api/src/Cardano/Api/SerialiseCBOR.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-api/src/Cardano/Api/SerialiseCBOR.hs:5]: We can rename the modules later if we want to. They're all internal, not exposed.

**Comment by dcoutts** [comment]
Added two more commits to merge TextView and SerialiseTextEnvelope modules, which also fixes an orphan instance warning.

**Pull request merged**